tests/sanity_helpers.py

```diff
@@ -1,133 +1,73 @@
+        node.wait_for_nodes_status()
+        ceph_cluster = CephCluster()
+        assert ceph_health_check(
+            namespace=config.ENV_DATA['cluster_namespace']
+        )
+        ceph_cluster.cluster_health_check(timeout=60)
 
+    def create_resources(self, pod_factory, run_io=True):
+        """
+        Sanity validation - Create resources (FS and RBD) and run IO
 
-def create_resources(resources, run_io=True):
-    """
-    Sanity validation - Create resources (FS and RBD) and run IO
+        Args:
+            pod_factory (function): A call to pod_factory function
+            run_io (bool): True for run IO, False otherwise
 
-    Args:
-        resources (tuple): Lists of projects, secrets, pools,
-            storageclasses, pvcs and pods
-        run_io (bool): True for run IO, False otherwise
+        """
+        logger.info(f"Creating resources and running IO as a sanity functional validation")
 
-    """
-    # Create resources and run IO for both FS and RBD
-    # Unpack resources
-    projects, secrets, pools, storageclasses, pvcs, pods = resources[:6]
-
-    # Project
-    projects.append(helpers.create_project())
-
-    # Secrets
-    secrets.append(helpers.create_secret(constants.CEPHBLOCKPOOL))
-    secrets.append(helpers.create_secret(constants.CEPHFILESYSTEM))
-
-    # Pools
-    pools.append(helpers.create_ceph_block_pool())
-    pools.append(helpers.get_cephfs_data_pool_name())
-
-    # Storageclasses
-    storageclasses.append(
-        helpers.create_storage_class(
-            interface_type=constants.CEPHBLOCKPOOL,
-            interface_name=pools[0].name,
-            secret_name=secrets[0].name
-        )
-    )
-    storageclasses.append(
-        helpers.create_storage_class(
-            interface_type=constants.CEPHFILESYSTEM,
-            interface_name=pools[1],
-            secret_name=secrets[1].name
-        )
-    )
-
-    # PVCs
-    pvcs.append(helpers.create_pvc(
-        sc_name=storageclasses[0].name, namespace=projects[0].namespace)
-    )
-    pvcs.append(helpers.create_pvc(
-        sc_name=storageclasses[1].name, namespace=projects[0].namespace)
-    )
-    for pvc in pvcs:
-        helpers.wait_for_resource_state(pvc, constants.STATUS_BOUND)
-        pvc.reload()
-
-    # Pods
-    pods.append(
-        helpers.create_pod(
-            interface_type=constants.CEPHBLOCKPOOL, pvc_name=pvcs[0].name,
-            namespace=projects[0].namespace
-        )
-    )
-    pods.append(
-        helpers.create_pod(
-            interface_type=constants.CEPHFILESYSTEM, pvc_name=pvcs[1].name,
-            namespace=projects[0].namespace
-        )
-    )
-    for pod in pods:
-        helpers.wait_for_resource_state(pod, constants.STATUS_RUNNING)
-        pod.reload()
-
-    if run_io:
-        # Run IO
-        for pod in pods:
+        for interface in [constants.CEPHBLOCKPOOL, constants.CEPHFILESYSTEM]:
+            self.pod_objs = pod_factory(interface)
```

Comment:
this is a mistake.
Will upload a commit soon to change this to:
            self.pod_objs.append(pod_factory(interface))